### DISCUSSION TITLE
connection to planetscale db from spark: where to add 'set workload =olap"

### DISCUSSION THREAD

**Discussion by Rache1C** [planetscale/discussion#378] [Q&A] [answered]
Hello, I have the following working connection which is reaching the 10k row read limit. Can anyone tell me where to set the workload to olap in this connection. Thanks.

`remote_table = (
        spark.read.format("jdbc")
        .option("driver", driver)
        .option("url", url)
        .option("user", username)
        .option("password", passwd)
        .option("ssl", True)
        .option("query", "select * from {}".format(table))
        .load()
    )`

**Comment by orware** [comment, marked as answer]
Hello @rashelie,

Please try adding the following onto the end of your Database URL, which I assume is already being passed in as part of the `url` option shown in your example:

```
sessionVariables=workload=olap
```

(Depending on whether your Database URL already includes a parameter or two at the end the above may need to be preceded by `?` or `&`.)

More details about the `sessionVariables` option can be found in the [MySQL Java Connector](https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-connp-props-session.html#cj-conn-prop_sessionVariables) documentation.

**Reply by mariomeissner** [reply]
Thank you, this worked for me. I am dumping data from spark into planetscale and no matter what batch settings I chose I ran into timeouts. Adding this parameter to the URL allowed me to finally complete my data dump.
